troubleshooting.md

```diff
@@ -0,0 +1,551 @@
+```
+
+and then issuing the journalctl command to inspect the logs:
+
+```
+[core@localhost ~]$ journalctl -b -f -u bootkube.service
+-- Logs begin at Thu 2019-12-05 10:23:22 UTC. --
+Feb 05 10:29:51 localhost bootkube.sh[3119]: https://etcd-0.kni7.cloud.lab.eng.bos.redhat.com:2379 is unhealthy: failed to commit proposal: context deadline exceeded
+Feb 05 10:29:51 localhost bootkube.sh[3119]: https://etcd-1.kni7.cloud.lab.eng.bos.redhat.com:2379 is unhealthy: failed to commit proposal: context deadline exceeded
+Feb 05 10:29:51 localhost bootkube.sh[3119]: Error: unhealthy cluster
+Feb 05 10:29:51 localhost podman[7743]: 2020-02-05 10:29:51.487999651 +0000 UTC m=+5.434743491 container died a62fd32f5b22e117e1fa5885ae956c5085f3ac5cc47fe88256b8282a9b45695d (image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:b3f6e79c60f60a3a8b8422adf82182de632ba2594b34a5b2e56e3564186efe77, name=etcdctl)
+```
+Once logged in to this VM, please check if all the pods(dnsmasq,mariadb,httpd, ironic pods etc) are in running state. 
+
+To check the logs of all the containers , please the below commands:
+
+```
+sudo podman logs ironic-api
+```
+You will get the error messages if the master nodes are not `powered on` by inspecting the ironic pods. 
+
+Once the ironic poewers on the master nodes, verify if RHCOS images are pushed to the cluster, once again inspect the logs of httpd and ironic pods logs from Bootstrap VM. 
+
+## BootstrapVM cannot download RHCOS image
+
+After the Bootstrap VM goes up comes a phase where podman raises 2 containers:
+
+```
+ipa-downloader
+coreos-downloader
+```
+
+These both containers will download resources from (in our case) the jump / bastion host since its a disconnected install.This is the relevant section where is this step performed, modify it in order to let the provisioner to access to the resources:
+
+```
+bootstrapOSImage: http://[<IPv4/Ipv6-IP-Registry]/rhcos-43.81.202001142154.0-qemu.x86_64.qcow2.gz?sha256=9d999f55ff1d44f7ed7c106508e5deecd04dc3c06095d34d36bf1cd127837e0c
+clusterOSImage: http://[<IPv4/Ipv6-IP-Registry>]/rhcos-43.81.202001142154.0-openstack.x86_64.qcow2.gz?sha256=a1bda656fa0892f7b936fdc6b6a6086bddaed5dafacedcd7a1e811abb78fe3b0
+```
+
+##Check the status of these containers using 
+
+```
+#podman logs -f ipa-downloader
+
+#podman logs -f coreos-downloader
+
+```
+After the installation is complete copy the config file to the `~/.kube/config` dir to interact with the cluster
+
+```
+cp /install-dir/ocp/auth/kubeconfig  ~/.kube/config
+```
+Or 
+
+```
+export KUBECONFIG=/install-dir/ocp/auth/kubeconfig
+```
+
+Verify if all the control plane nodes are up and running 
+
+```
+oc get nodes
+NAME                                         STATUS   ROLES           AGE    VERSION
+master-0.cloud.lab.eng.bos.redhat.com   Ready    master,worker   4h   v1.16.2
+master-1.cloud.lab.eng.bos.redhat.com   Ready    master,worker   4h   v1.16.2
+master-2.cloud.lab.eng.bos.redhat.com   Ready    master,worker   4h   v1.16.2
```

Comment:
its troubleshooting section, show how about instead showing if its not working?
Example 
```
$ oc get nodes
error: the server doesn't have a resource type "nodes"
```
Why did I get the above? What should I do to solve this?

Reply:
This is explained in the immediate next paragraph, which helps solve the Node related problems.